### ISSUE TITLE
Always opens in full screen mode

### ISSUE THREAD

**Issue by sameer-coder** [SpartanJ/ecode#2]
Thanks for the fantastic editor. I am on windows 10 and using the windows version. The editor for some reason only works in full screen mode even if full screen mode is disabled in settings.

**Comment by SpartanJ** [comment]
That's very weird. Can you please share the application logs at: `C:\Users\{username}\AppData\ecode\`. Or better, zip that folder that contains the screen configuration and logs. Thanks!

**Comment by redstoneleo** [comment]
Yes, the Windows taskbar wad hidden.
I don't know where the settings resides, so I'd zip the folder containing  ecode.exe and uploaded it here.
[ecode.zip](https://github.com/SpartanJ/ecode/files/10435288/ecode.zip)

**Comment by redstoneleo** [comment]
There is no such path `C:\Users\{username}\AppData\ecode\` even replacing {username} with my Windows user name.
![image](https://user-images.githubusercontent.com/5896180/212937793-026bd809-e554-4a9e-817d-cd7096b3b309.png)

**Comment by SpartanJ** [comment]
Oh, ok, the path sometimes is kind of weird, but you can read it from the console, if you can help me do the following:
Press F12 to show the console (or Window -> Show console), and scroll all the way up (shift + home, or with the mouse wheel), the second line is the load of the config file, you'll be able to see the full path of the `config.cfg` file there.
Thanks!

**Comment by redstoneleo** [comment]
Content of `C:\Users\ueserName\AppData\Roaming\ecode\config.cfg`
```

[editor]
colorscheme=eepp
show_line_numbers=1
show_white_spaces=1
highlight_matching_brackets=1
highlight_current_line=1
vertical_scrollbar=1
horizontal_scrollbar=1
font_size=11.00dp
auto_close_brackets=
highlight_selection_match=1
color_picker_selection=1
color_preview=1
minimap=1
show_doc_info=1
hide_tab_bar_on_single_tab=0
single_click_tree_navigation=0
sync_project_tree_with_editor=1
auto_close_xml_tags=1
line_spacing=0.00dp
cursor_blinking_time=500.00ms

[ui]
font_size=11.00dp
show_side_panel=0
panel_position=left
serif_font=fonts/NotoSans-Regular.ttf
monospace_font=fonts/DejaVuSansMono.ttf
terminal_font=fonts/DejaVuSansMonoNerdFontComplete.ttf
fallback_font=fonts/DroidSansFallbackFull.ttf
ui_color_scheme=light

[document]
trim_trailing_whitespaces=0
force_new_line_at_end_of_file=0
auto_detect_indent_type=1
write_bom=0
indent_width=4
indent_spaces=0
windows_line_endings=0
tab_width=4
line_breaking_column=100

[search_bar]
case_sensitive=0
lua_pattern=0
whole_word=0
escape_sequence=0

[global_search_bar]
case_sensitive=0
lua_pattern=0
whole_word=0
escape_sequence=0

[terminal]
font_size=11.00dp
colorscheme=eterm

[window]
vsync=1
glversion=OpenGL 2
multisamples=0
frameratelimit=0

[workspace]
restore_last_session=0

[plugins]
autocomplete=1
autoformatter=1
linter=1
lspclient=1


```

**Comment by SpartanJ** [comment]
Awesome, can you please share `state.cfg` that it`s in the same directory? Thanks again

**Comment by redstoneleo** [comment]
The content of `C:\Users\userName\AppData\Roaming\ecode\state.cfg`
```

[window]
width=1920
height=1080
maximized=0
pixeldensity=1.500000
panel_partition=15.00%
display_index=0
x=0
y=0

[files]
recentfiles=

[folders]
recentfolders=C%3A%5CUsers%5C22815%5CDownloads%5Cecode


```

**Comment by SpartanJ** [comment]
Thanks, well, I don't see anything interesting. I'm gonna request you the logs that it's located in the same folder, it's called `ecode.log`. So I can see your basic PC info and see if I can reproduce it somehow. Thanks again.

**Comment by redstoneleo** [comment]
All content of `C:\Users\userName\AppData\Roaming\ecode`:
[ecode.zip](https://github.com/SpartanJ/ecode/files/10436613/ecode.zip)

**Comment by SpartanJ** [comment]
You are the man!
Last request if possible:
Can you download the last SDL2 version from [here](https://github.com/libsdl-org/SDL/releases/download/release-2.26.2/SDL2-2.26.2-win32-x64.zip). Open the zip file, and replace the SDL2.dll file that comes with ecode for the one that it's in that zip file? This looks like an SDL2 bug, and ecode it's not using the latest version, that might help.

**Comment by redstoneleo** [comment]
BTW, Chinese characters in the editor become garbled, for example 
, `编辑器` in it became the following
 
![image](https://user-images.githubusercontent.com/5896180/212944298-45fcb51a-c07c-4326-9685-4a8fd5d02224.png)

I guess a lot of bugs should be fixed in future.

**Comment by SpartanJ** [comment]
> It didn't help.

I think that I understand the issue. The bug is that it's starting with an incorrect resolution when the screen DPI is bigger than 105 because I hardcoded the initial resolution depending on the screen DPI, and with >= 105 DPI it will create by default a bigger window in 1080p. It's an easy fix.
May be you can force the resolution by editing the `state.cfg` file with other editor. You can try to set 1280x720.
Like:
```
[window]
width=1280
height=720
maximized=0
pixeldensity=1.500000
panel_partition=15.00%
display_index=0
x=0
y=0
```


> BTW, Chinese characters in the editor become garbled, for example, 编辑器 in it became the following

Please read:
https://github.com/SpartanJ/ecode/issues/7#issuecomment-1382669178
You'll need to use a Monospaced font with Chinese support.

**Comment by SpartanJ** [comment]
I was able to reproduce it and fix it. If anyone wants to try it out here's a release with the fix applied:
[ecode-windows-0.4.1-x86_64.zip](https://github.com/SpartanJ/ecode/files/10441363/ecode-windows-0.4.1-x86_64.zip)

**Comment by redstoneleo** [comment]
> I was able to reproduce it and fix it. If anyone wants to try it out here's a release with the fix applied: [ecode-windows-0.4.1-x86_64.zip](https://github.com/SpartanJ/ecode/files/10441363/ecode-windows-0.4.1-x86_64.zip)

It really got fixed, thanks for your effort!
